lib/spack/spack/bootstrap.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import contextlib
+import os
+import sys
+
+import llnl.util.filesystem as fs
+import llnl.util.tty as tty
+
+import spack.spec
+import spack.store
+import spack.user_environment as uenv
+import spack.util.executable
+from spack.util.environment import EnvironmentModifications
+
+
+@contextlib.contextmanager
+def system_python_context():
+    python_cls = type(spack.spec.Spec('python').package)
+    python_prefix = os.path.dirname(os.path.dirname(sys.executable))
+    externals = python_cls.determine_spec_details(
+        python_prefix, [os.path.basename(sys.executable)])
+    external_python = externals[0]
```

Comment:
This has weird side effects if the code is ever hit with unit tests. The tests based on `builtin.mock` will fail with:
```
    @contextlib.contextmanager
    def system_python_context():
        python_cls = type(spack.spec.Spec('python').package)
        python_prefix = os.path.dirname(os.path.dirname(sys.executable))
>       externals = python_cls.determine_spec_details(
            python_prefix, [os.path.basename(sys.executable)])
E       AttributeError: type object 'Python' has no attribute 'determine_spec_details'
```
since they'll pick the Python from `builtin.mock`. Other tests are instead installing in the user store:
```console
$ spack find
[ ... ]
-- test-debian6-x86_64 / gcc@4.5.0 ------------------------------
libc@1.0
```